clusterloader2/testing/load/config.yaml

```diff
@@ -50,14 +53,22 @@
 {{$mediumDeploymentsPerNamespace := DivideInt $podsPerNamespace (MultiplyInt 4 $MEDIUM_GROUP_SIZE)}}
 # smallDeployments - 1/2 of namespace pods should be in small Deployments.
 {{$smallDeploymentsPerNamespace := DivideInt $podsPerNamespace (MultiplyInt 2 $SMALL_GROUP_SIZE)}}
-# Reduce the number of small and medium deployments per namespace
-{{$smallDeploymentsPerNamespace := SubtractInt $smallDeploymentsPerNamespace $SMALL_STATEFUL_SETS_PER_NAMESPACE}}
-{{$mediumDeploymentsPerNamespace := SubtractInt $mediumDeploymentsPerNamespace $MEDIUM_STATEFUL_SETS_PER_NAMESPACE}}
 
-# Reduce the number of small, medium, big deployments per namespace.
-{{$smallDeploymentsPerNamespace := SubtractInt $smallDeploymentsPerNamespace 1}}
-{{$mediumDeploymentsPerNamespace := SubtractInt $mediumDeploymentsPerNamespace 1}}
-{{$bigDeploymentsPerNamespace := SubtractInt $bigDeploymentsPerNamespace 1}}
+# Stateful sets are enabled. Reduce the number of small and medium deployments per namespace
+# See https://github.com/kubernetes/perf-tests/issues/1036#issuecomment-607631768
+# Ensure non zero or negative after subtraction.
+{{$smallDeploymentsPerNamespace := MaxInt 1 (SubtractInt $smallDeploymentsPerNamespace $SMALL_STATEFUL_SETS_PER_NAMESPACE)}}
+{{$mediumDeploymentsPerNamespace := MaxInt 1 (SubtractInt $mediumDeploymentsPerNamespace $MEDIUM_STATEFUL_SETS_PER_NAMESPACE)}}
+
+# Jobs are enabled. Reduce the number of small, medium, big deployments per namespace.
+# Ensure non zero or negative after subtraction.
+{{$smallDeploymentsPerNamespace := MaxInt 1 (SubtractInt $smallDeploymentsPerNamespace 1)}}
+{{$mediumDeploymentsPerNamespace := MaxInt 1 (SubtractInt $mediumDeploymentsPerNamespace 1)}}
+{{$bigDeploymentsPerNamespace := MaxInt 1 (SubtractInt $bigDeploymentsPerNamespace 1)}}
```

Comment:
We should have: `MaxInt 0 ...` instead of `MaxInt 1 ...`. Here and above. Then we won't need lines 69-71

Reply:
I added :69-71 because there is nothing limiting `$bigDeploymentsPerNamespace` above any more, it will be `250 * 4 - 1 = 999` on this line

Reply:
Hmm, the `bigDeploymentsPerNamespace` is calculated as
```
{{$bigDeploymentsPerNamespace := DivideInt $podsPerNamespace (MultiplyInt 4 $BIG_GROUP_SIZE)}}
```
And if we have `$namespaces=1` then `$podsPerNamespace = 30 * #Nodes`

So, e.g. in a 10 node cluster
```
bigDeploymentsPerNamespace := 30* 10 / 4 / 250 = 0
```

I believe it should work, we don't need any custom logic there

Reply:
Ah did I miss something. Let me try again later. I vaguely recall it wasn't working as expected when I set it to  MaxInt 0

Reply:
Ok it works! Made the change accordingly